src/parser/serializer.ts

```diff
@@ -783,7 +797,7 @@ export function serializeInto(
       } else if (type === 'number') {
         index = serializeNumber(buffer, key, value, index);
       } else if (type === 'bigint' || isBigInt64Array(value) || isBigUInt64Array(value)) {
```

Comment:
The condition left behind here checks for more than it should, this should just be `type === 'bigint'`